apps/institution-service/src/shared/utils/permissionValidation.ts

```diff
@@ -88,25 +88,46 @@ export enum EditAggregatorIntegrationValidationErrorReason {
   NotYourAggregator,
 }
 
-export const validateUserCanEditAggregatorIntegration = async ({
+export enum UserAction {
+  EDIT,
+  DELETE,
+}
+
+export const validateUserCanActOnAggregatorIntegration = async ({
```

Comment:
This works, but it could be more extensible and IMO simpler.

I would do this:

`
export const createValidateUserCanActOnAggregatorIntegration = ({
  adminPermission,
  aggregatorPermission,
}) => async ({
  aggregatorIntegrationId,
  req
}) => {}

export const validateUserCanEditAggregatorIntegration = createValidateUserCanActOnAggregatorIntegration({
  adminPermission: UiUserPermissions.UPDATE_AGGREGATOR_INTEGRATION,
  aggregatorPermission: UiUserPermissions.UPDATE_AGGREGATOR_INTEGRATION_AS_AGGREGATOR
})
`